### ISSUE TITLE
Hops limit partial results not returned

### ISSUE THREAD

**Issue by antejavor** [memgraph/memgraph#2175]
**Memgraph version**
2.18 
```
CREATE INDEX ON :Node(id);
FOREACH (i IN range(0,10) | CREATE (:Node {id: i}));
match (n1:Node),(n2:Node) where n1.id=n2.id-1 create (n1)-[:PARENT]->(n2);

MATCH path=(n)-[r]-(m) 
RETURN path

USING HOPS LIMIT 5 MATCH path=(n:Node{id: 0})-[*BFS]->(m:Node {id: 7})
RETURN path;
```

I would expect this to return results.

**Comment by DavIvek** [comment]
From my perspective, this isn't a bug, but maybe we can document it better.
Hops limit won't give you partial results since the query tries to find a path from Node with id 1 to Node with id 7. It can't get from Node with id 1 to Node with id 7 in 5 hops, so the result is the empty set.
If you write a query like this:
`USING HOPS LIMIT 5 MATCH path=(n:Node{id: 0})-[*BFS]->(m:Node) return path;`
you'll get 5 paths as a result of the query.

**Comment by DavIvek** [comment]
@antejavor Can we close this?

**Comment by antejavor** [comment]
@DavIvek, this means if you have a specific target end node, the hops feature is not working? 

If I know a start node and want to get to the end node but do not want to hop too deep and want to use the HOPS limit, this won't work.  

To me, this is fine, but let's write that example/explanation on the docs, so the actually issue can be moved to the docs repository to document a bit better. I will transfer it, just confirm if I am assuming this correctly.

**Comment by DavIvek** [comment]
@antejavor Yes, your assumption is correct, It won't return you any results, but that is how we specified the feature. Maybe we can add that in next version of the feature.
